drivers/serial/Kconfig.mcux_flexcomm

```diff
@@ -10,6 +10,7 @@ config UART_MCUX_FLEXCOMM
 	select SERIAL_SUPPORT_INTERRUPT
 	select SERIAL_SUPPORT_ASYNC if \
 		DT_HAS_NXP_LPC_DMA_ENABLED
+	select PM_POLICY_DEVICE_CONSTRAINTS if PM
```

Comment:
The title references RW612 where this gets set already if PM is enabled (https://github.com/zephyrproject-rtos/zephyr/blob/fbdf4b9c64f66fea7ba9ab1c473fa6f19f8735a0/soc/nxp/rw/Kconfig.defconfig#L74)

But I'm guessing that since this is shared IP we need this here. We might need to think through a bit this config and why we set it in the RW defconfig... yet still need to set it in the drivers Kconfig

Reply:
Yes, I put this here because it is not specific to RW, even though RW is the only platform right now in tree that is enabling PM deep enough to disable the UART. But any platform that disables the UART on low power mode will need this dependency to avoid the garbage character.